app/status.go

```diff
@@ -315,6 +315,34 @@ func (a *App) SetStatusDoNotDisturb(userId string) {
 	a.Publish(event)
 }
 
+func (a *App) SetStatusOutOfOffice(userId string) {
+	if !*a.Config().ServiceSettings.EnableUserStatuses {
+		return
+	}
+
+	status, err := a.GetStatus(userId)
+
+	if err != nil {
+		status = &model.Status{UserId: userId, Status: model.STATUS_OUT_OF_OFFICE, Manual: false, LastActivityAt: 0, ActiveChannel: ""}
```

Comment:
This doesn't seem like the right way to handle errors here.

Reply:
I agree, just following MM conventions, check `setStatusOnline`
* https://github.com/mattermost/mattermost-server/blob/master/app/status.go#L177

Looks like if the `GetStatus` method returns an error more than likely there was no status in the database because at that point the error returns from the database layer.

Therefore just reset to the intended status.